services/fileuploader/fileuploader.go

```diff
@@ -173,8 +173,8 @@ func overrideWithSettings(defaultConfig map[string]interface{}, settings backend
 	for k, v := range settings.Config {
 		config[k] = v
 	}
-	if _, ok := config["externalId"]; ok && settings.Type == "S3" {
-		config["externalId"] = workspaceID
+	if settings.Type == "S3" {
```

Comment:
Shouldn't we be doing something similar to that, i.e. check for the presence of `iamRoleArn` config property?
https://github.com/rudderlabs/rudder-server/blob/master/services/filemanager/filemanager.go#L102-L105